src/frontend/src/session.rs

```diff
@@ -752,6 +762,17 @@ impl Session<PgResponseStream> for SessionImpl {
     fn id(&self) -> SessionId {
         self.id
     }
+
+    fn end_query(&self, value_stream: &PgResponseStream) -> PsqlResult<()> {
```

Comment:
Can we make `end_query` a method just for `SessionImpl`? Because not all impls of `Session` are stateful in terms of queries.

Also, is it possible to move the call of `end_query` to `DistributedQueryStream::drop`, so that we can leave pgwire untouched.